### PR TITLE
Client Code Guidelines put into docs

### PR DESCRIPTION
**Pull request by me-andre**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

**Comment by techsmyth** [comment]
@me-andre Nice set of guidelines. 

There are a few other points I would like to discuss: 
- Usage of models to isolate from grapgql type changes: we should in general be minimizing the direct usage of types from the generated graphql
- Only lookup based on uuid, not nameid: I picked up before that callouts are looking up based on NameID, which is not supported via lookups so the client should not be using nameID to find a particular child
- Directory structure  in domain: we need a review of the structure being used within src/domain. This has some strange nesting e.g. around templates / innovation library. Templates exist independently of Innovation Library.

Looking forward to discussing more.

File: docs/code-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+  );
+};
+```
+
+## Component Library and Avoidance of code repetition
+
+Any component that represents a self-sufficient UI building block and/or contains CSS tweaks, should be considered a UI Library Component and put into `src/core/ui`.
+Domain-specific components should normally not include low-level markup (such as rendering `<div>` or `<Box>`) or CSS with the exception of one-time customization/an exclusive override for a specific page.
+
+## To be done:
```
Comment (ccanos) [inline docs/code-guidelines.md:306]: I would like to have some kind of summary of all of this to take a look during CodeReviews, like a list of checks or something like that.
I guess somebody else can do it, probably myself :D

**Review by ccanos — CHANGES_REQUESTED** [review]

File: docs/code-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+  - `src/domain/challenge/ChallengeDashboardPage.tsx`
+  - `src/domain/collaboration/callout/calloutView/CalloutView.tsx`
+- `src/platform` contains the code that doesn't directly belong to the business domain but is specific to the app, like `PlatformFooter` or `TopLevelDesktopLayout` component:
+  - `src/platform/routes/RedirectToLanding.tsx`
+  - `src/platform/ui/platformFooter/PlatformFooter.tsx`
+
+> :warning: Please keep in mind the difference between `src/platform` and `src/domain/platform`. The latter is for placing
+> the code related to entities that belong to the Platform domain of Alkemio.
+
+> :warning: All code built for Admin side should go under `src/platform/admin`.
```
[the remark this answers is not in the document]
Comment (ccanos) [reply docs/code-guidelines.md:22]: you mentioned in your comment, " Admin prefix for admin-side pages, views and containers". I didn't see that clearly in the document

**Review by ccanos — APPROVED** [review]

**Comment by ccanos** [comment]
closed to retrigger CLA

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**